### PR TITLE
Selectors

### PR DESCRIPTION
**Pull request by SiansandaTamara**
###Selectors###
-Use of immune state
-Use of useselect
- Use of dispatch
- No linter errors
- Use of redux

### CODE REVIEW

**Review by DeliceLydia — COMMENTED** [review]
## Hi @SiansandaTamara, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that did not follow git workflow please open your PR from feature branch to develop not main branch

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by OLIPLICHE — COMMENTED** [review]
Hi <STUDENT_MENTION>, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that did not follow git workflow please open your PR from `feature` branch to `develop` not `main` branch

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/Redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+const ADD_BOOK = 'books/ADD_BOOK';
+const REMOVE_BOOK = 'books/REMOVE_BOOK';
+// const GET_BOOKS = 'books/GET_BOOKS';
+
```
Comment (Gambit142) [inline src/Redux/books/books.js]: - [ ] Kindly remove all unnecessary commented-out code in this file. Removing such codes makes your code clean and it shows that you adopt best JavaScript practices when building an application. 👍

File: src/Redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    category: '',
+  },
+];
+
+// Actions
+
+export const addBook = (book) => ({ type: ADD_BOOK, book });
+
+// export const getBook = (payload) => ({ type: ADD_BOOK, payload });
+
```
Comment (Gambit142) [inline src/Redux/books/books.js]: - [ ] Kindly remove all unnecessary commented-out code in this file. Removing such codes makes your code clean and it shows that you adopt best JavaScript practices when building an application. 👍

File: src/Redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      const bookObj = {
+        id: state.length + 1,
+        author: action.book.author,
+        title: action.book.title,
+      };
+      return [...state, bookObj];
+    }
+
+    // case GET_BOOKS:
+    //   return [...state, action.payload];
```
Comment (Gambit142) [inline src/Redux/books/books.js]: - [ ] Kindly remove all unnecessary commented-out code in this file. Removing such codes makes your code clean and it shows that you adopt best JavaScript practices when building an application. 👍

File: src/components/BookContainer.js [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
+            {' by '}
+            {book.author}
+            {' - '}
+            {book.category}
+          </li>
+          <button type="button" id={book.id} onClick={
+            () => dispatch(deleteBook(book.id))
+          }>Remove</button>
+        </div>
+      ))} */}
```
Comment (Gambit142) [inline src/components/BookContainer.js]: - [ ] Kindly remove all unnecessary commented-out code in this file. Removing such codes makes your code clean and it shows that you adopt best JavaScript practices when building an application. 👍

File: src/Redux/categories/categories.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+const CHECK_STATUS = 'CHECK_STATUS';
+
+export const checkStatus = () => ({
+  type: CHECK_STATUS,
+});
```
Comment (Gambit142) [inline src/Redux/categories/categories.js]: - [ ] In the `Categories` component, kindly create a function that dispatches this action creator and include JSX in the component that displays the **_under construction_** string when the `Check Status` button is clicked. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @SiansandaTamara 👋, 

Good job so far!

## Highlights
- The redux `store` is available to all components ✔️ 
- The components dispatch their appropriate actions when an action is triggered ✔️ 
- The list of books is displayed from the redux store ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by uwadonat — APPROVED** [review]
Hi @SiansandaTamara,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
appr2

# Status: Approved ✔️✔️✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- To use correctly gitflow. In this flow the pull request compare feature branch with development branch, not main branch. you can refer to this [documentation.](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**